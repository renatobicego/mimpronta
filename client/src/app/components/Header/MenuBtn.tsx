"use client"
import Image from "next/image"
import Link from "next/link"
import React, { useEffect, useState } from "react"

const links = [
    {
        href: '/',
        text: 'Inicio'
    },
    {
        href: '/#servicios',
        text: 'Servicios'
    },
    {
        href: '/blog',
        text: 'Blog'
    },
    {
        href: '/sobreMimpronta',
        text: 'Sobre Mimpronta'
    },
]

const MenuBtn = () => {
    const [open, setOpen] = useState(false)
    const [animationOpenComplete, setAnimationOpenComplete] = useState(false);
    const [shrinkNav, setShrinkNav] = useState(false);

    // useEffect(() => {
    //     if(!open && shrinkNav){
    //         setShrinkNav(false)
    //     }
    // }, [shrinkNav, open])

    const handleAnimationEnd = () => {
        // Esta función se ejecutará al finalizar la animación
        setAnimationOpenComplete(true);
    };

    const handleOpen = () => {
        setOpen(true)
        setAnimationOpenComplete(false)
    }

    const handleClose = () => {
        setShrinkNav(true);
        
        setTimeout(() => {
            setOpen(false);
            setShrinkNav(false);
            setAnimationOpenComplete(false)
        }, 200);
    }


    if(open){
        return (
            <>
                <nav role="navigation" className={`hidden lg:flex-between py-1 px-3 xsm:py-1.5 xsm:px-4 lg:pl-12 lg:pr-6 
                    border-2 border-negro rounded-[10px] lg:rounded-2xl 
                    gap-2 lg:gap-10 font-semibold text-sm xsm:text-base
                    ${shrinkNav ? 'menu-animation-close' : 'menu-animation-open'}`}
                    onAnimationEnd={handleAnimationEnd}>
                    {links.map((link, i) => {
                        return (
                            <li onClick={handleClose} key={i} className={`opacity-0 list-none ${animationOpenComplete && !shrinkNav ? 'animate-opacity' : ''}`}>
                                <a href={link.href}>
                                    {link.text}
                                </a>
                            </li>
                        )
                    })}
                    <Image 
                        src={'/icons/close.png'} 
                        alt="icono menú" 
                        onClick={handleClose}
                        className="w-3 h-3 xsm:w-4 xsm:h-4 cursor-pointer mr-1 menu-animation-open"
                        width={50} 
                        height={50}/>
                </nav>
                <div className="w-screen h-screen absolute left-0 top-0 backdrop-blur-sm lg:hidden"></div>
                <nav 
                    className={`h-screen w-screen xsm:w-[60vw] sm:w-[40vw] absolute top-0 left-0 xsm:left-[40vw] sm:left-[60vw] 
                                flex lg:hidden flex-col bg-negro text-white
                                pt-[4vh] sm:pt-[5.8vh] pl-10 pr-[9vw] xsm:px-[9vw] sm:px-[7vw]  gap-4
                                ${shrinkNav ? 'hg-menu-animation-close' : 'hg-menu-animation-open'} `}
                    onAnimationEnd={handleAnimationEnd}>
                    <Image 
                        src={'/icons/closeWhite.png'} 
                        alt="icono menú" 
                        onClick={handleClose}
                        className="w-3 h-3 xsm:w-4 xsm:h-4 cursor-pointer mr-1 self-end"
                        width={50} 
                        height={50}/>
                    {links.map((link, i) => {
                        return (
                            <li onClick={handleClose} key={i} className={`list-none font-semibold`}>
                                <a href={link.href}>
                                    {link.text}
                                </a>
                            </li>
                        )
                    })}
                </nav>
            </>
        )
    }
    return (
        <button className={`flex-between py-1 px-3 xsm:py-1.5 xsm:px-4 xl:px-6 border-2 
            border-negro rounded-[10px] lg:rounded-2xl 
            gap-2 lg:gap-4 font-semibold text-sm xsm:text-base`}
            onClick={handleOpen}>
            <span className={`pb-0.5 lg:pb-0`}>
                Menú
            </span>
            <Image 
                src={'/icons/hgMenu.png'} 
                alt="icono menú" 
                className="w-5 h-5 xsm:w-6 xsm:h-6"
                width={50} 
                height={50}/>
        </button>
    )
}

export default MenuBtn