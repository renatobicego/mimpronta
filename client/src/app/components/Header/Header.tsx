"use client";
import Image from "next/image";
import MenuBtn from "./MenuBtn";
import useScrollPosition from "@/utils/hooks/useScrollPosition";
import Link from "next/link";
import { usePathname } from "next/navigation";

const Header = () => {
  const { y: scrollY } = useScrollPosition();

  const opacity = Math.min(Math.round(scrollY / 1.5), 100);
  const pathname = usePathname();

  const headerStyle = {
    backgroundColor: `rgba(255, 255, 255, ${
      pathname === "/" ? opacity : 100
    }%)`,
    // backgroundColor: `rgba(255, 255, 255, ${opacity}%)`,
  };

  return (
    <header
      style={headerStyle}
      className={`fixed top-0 flex-between left-0 z-50
        w-screen px-[5%] lg:px-[8.335%] h-14 py-[5vh] lg:py-[6.5vh]  font-text
        bg-white`}
    >
      <a
        href={"/#inicio"}
        className="w-1/3 sm:w-1/4 md:w-1/5 lg:w-1/6 2xl:w-[14%]"
      >
        <Image
          src={"/logo-02.png"}
          alt="logo"
          className="w-full"
          width={200}
          height={200}
        />
      </a>
      <MenuBtn />
    </header>
  );
};

export default Header;
