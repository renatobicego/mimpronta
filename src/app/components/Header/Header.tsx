"use client"
import Image from "next/image"
import MenuBtn from "./MenuBtn"
import useScrollPosition from "@/utils/hooks/useScrollPosition"

const Header = () => {
  const {y: scrollY} = useScrollPosition()

  const opacity = Math.min(Math.round(scrollY / 1.5), 100);

  const headerStyle = {
    backgroundColor: `rgba(255, 255, 255, ${opacity}%)`,
  };

  return (
    <header 
      style={headerStyle}
      className={`fixed top-0 flex-between left-0 z-50
        w-screen px-[5%] lg:px-[8.335%] h-14 py-[5vh] sm:py-[6.5vh]  font-text
        bg-white`}>
        <Image 
          src={'/logo-02.png'} 
          alt="logo" 
          className="w-1/3 sm:w-1/4 md:w-1/5 lg:w-1/6 2xl:w-[14%]"
          width={200} 
          height={200}/>
        <MenuBtn />
    </header>
  )
}


export default Header