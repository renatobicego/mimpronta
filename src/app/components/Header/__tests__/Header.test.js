import { render, screen } from '@testing-library/react';
import Header from '../Header.tsx'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event'

describe('Header', () => {
    it('renders unchanged', () => {
        const {container} = render(<Header />)
        expect(container).toMatchSnapshot()
    })

    it('renders content', () => {
        render(<Header />)
        const logo = screen.getByAltText('logo')
        const menuBtn = screen.getByText('Menú')
        const navMenu = screen.queryByRole('navigation')

        expect(logo).toBeInTheDocument()
        expect(menuBtn).toBeInTheDocument()
        expect(navMenu).toBeNull()
    })

    it('renders nav menu', async() => {
        render(<Header />)
        const menuBtn = screen.getByText('Menú')
        await userEvent.click(menuBtn)

        await new Promise(resolve => setTimeout(resolve, 1000));

        const inicioLink = screen.getByText('Inicio');
        expect(inicioLink).toBeInTheDocument();
    })
});
