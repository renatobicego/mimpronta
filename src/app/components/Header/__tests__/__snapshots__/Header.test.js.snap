// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders unchanged 1`] = `
<div>
  <header
    class="fixed top-0 flex-between left-0 
        w-screen px-[5%] lg:px-[8.335%] h-14 py-[5dvh] sm:py-[6.5dvh] lg:py-[8dvh] font-text
        bg-white"
    style="background-color: rgba(255, 255, 255, 0);"
  >
    <img
      alt="logo"
      class="w-1/3 sm:w-1/4 md:w-1/5 lg:w-1/6 2xl:w-[14%]"
      data-nimg="1"
      decoding="async"
      height="200"
      loading="lazy"
      src="/_next/image?url=%2Flogo-02.png&w=640&q=75"
      srcset="/_next/image?url=%2Flogo-02.png&w=256&q=75 1x, /_next/image?url=%2Flogo-02.png&w=640&q=75 2x"
      style="color: transparent;"
      width="200"
    />
    <button
      class="flex-between py-1 px-3 xsm:py-1.5 xsm:px-4 xl:px-6 border-2 
            border-negro rounded-[10px] lg:rounded-2xl 
            gap-2 lg:gap-4 font-semibold text-sm xsm:text-base"
    >
      <span
        class="pb-0.5 lg:pb-0"
      >
        Menú
      </span>
      <img
        alt="icono menú"
        class="w-5 h-5 xsm:w-6 xsm:h-6"
        data-nimg="1"
        decoding="async"
        height="50"
        loading="lazy"
        src="/_next/image?url=%2Ficons%2FhgMenu.png&w=128&q=75"
        srcset="/_next/image?url=%2Ficons%2FhgMenu.png&w=64&q=75 1x, /_next/image?url=%2Ficons%2FhgMenu.png&w=128&q=75 2x"
        style="color: transparent;"
        width="50"
      />
    </button>
  </header>
</div>
`;
